ibis/expr/types/relations.py

```diff
@@ -739,8 +757,9 @@ def __getattr__(self, key: str) -> ir.Column:
         """
         try:
             return ops.Field(self, key).to_expr()
-        except com.IbisTypeError:
-            pass
+        except com.FieldNotFoundError as e:
```

Comment:
There is a slight difference in ux here that I'd love to unify
- if I do `Table.totally_bogus`, then I get `AttributeError: 'Table' object has no attribute 'bogus'`
- if I do `Table["totally_bogus"]`,  then I get `FieldNotFoundError: 'bogus' not found in Table object. Possible options: {'x'}`

I'm not sure which is better. If we say `Possible options:...` then that only includes the field names, and misses all the Table methods. But, all methods are 1. in the docs and 2. should have tab completion in many cases, so I bet typos are a lot more likely on column typos than method typos. So I think the FieldNotFoundError with the suggestion might be better.

Reply:
Also a difference between Tables and Structs. I'd love for them to have the same behavior too.